app/Models/SnipeSCIMConfig.php

```diff
@@ -0,0 +1,170 @@
+            
+            'validations' => [
+    
+                'urn:ietf:params:scim:schemas:core:2.0:User:userName' => 'required',
+                'urn:ietf:params:scim:schemas:core:2.0:User:password' => 'nullable',
+                'urn:ietf:params:scim:schemas:core:2.0:User:active' => 'boolean',
+                'urn:ietf:params:scim:schemas:core:2.0:User:emails' => 'required|array',
+                'urn:ietf:params:scim:schemas:core:2.0:User:emails.*.value' => 'required|email',
+                'urn:ietf:params:scim:schemas:core:2.0:User:roles' => 'nullable|array',
+                'urn:ietf:params:scim:schemas:core:2.0:User:roles.*.value' => 'required',
+    
+            ],
+    
+            'singular' => 'User',
+            'schema' => [Schema::SCHEMA_USER],
+    
+            //eager loading
+            'withRelations' => [],
+            'map_unmapped' => true,
+            'unmapped_namespace' => 'urn:ietf:params:scim:schemas:laravel:unmapped',
+            'description' => 'User Account',
+            
+            // Map a SCIM attribute to an attribute of the object.
+            'mapping' => [
+                
+                'id' => AttributeMapping::eloquent("id")->disableWrite(),
+                
+                'externalId' => null,
+                
+                'meta' => [
+                    'created' => AttributeMapping::eloquent("created_at")->disableWrite(),
+                    'lastModified' => AttributeMapping::eloquent("updated_at")->disableWrite(),
+                    
+                    'location' => (new AttributeMapping())->setRead(
+                        function ($object) {
+                            return route(
+                                'scim.resource',
+                                [
+                                'resourceType' => 'Users',
+                                'resourceObject' => $object->id
+                                ]
+                            );
+                        }
+                    )->disableWrite(),
+                    
+                    'resourceType' => AttributeMapping::constant("User")
+                ],
+                
+                'schemas' => AttributeMapping::constant(
+                    [
+                    'urn:ietf:params:scim:schemas:core:2.0:User',
+                    'example:name:space',
+                    ]
+                )->ignoreWrite(),
+                
+                // 'example:name:space' => [ //not really sure here?
+                //     'cityPrefix' => AttributeMapping::eloquent('cityPrefix')
+                // ],
+                
+                'urn:ietf:params:scim:schemas:core:2.0:User' => [
+                    
+                    'userName' => AttributeMapping::eloquent("username"),
+                    
+                    'name' => [
+                        'formatted' => null, // Snipe-IT only has first_name, last_name, not "name" or "fullname"
```

Comment:
We do have a (presenter?) method to get a full name - not sure that matters for these purposes. It’s not a citizen, just a presenter, but we have a way to pull a full name from first/last.

Reply:
We can do a shit version of it by `split()` now, but we’re definitely going to have a bad time with some more complex non-American names that have 3 or more names in them - we won’t know where to split them. But that’s at least a start, and I’m okay with it. I’ll mark this one as resolved.